tests/dryrun_test.go

```diff
@@ -0,0 +1,824 @@
+			Expect(err).To(BeNil())
+
+			By("Make a Dry-Run request to update the migration")
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				vmim, err = virtClient.VirtualMachineInstanceMigration(vmim.Namespace).Get(vmim.Name, &metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				vmim.Annotations = map[string]string{
+					"key": "42",
+				}
+				return tests.DryRunUpdate(restClient, resource, vmim.Name, vmim.Namespace, vmim, nil)
+
+			})
+
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Check that no update actually took place")
+			vmim, err = virtClient.VirtualMachineInstanceMigration(vmim.Namespace).Get(vmim.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vmim.Annotations["key"]).ToNot(Equal("42"))
+		})
+
+		It("patch a migration", func() {
+			By("Create a migration")
+			vmim, err = virtClient.VirtualMachineInstanceMigration(vmim.Namespace).Create(vmim)
+			Expect(err).To(BeNil())
+
+			By("Make a Dry-Run request to patch the migration")
+			patch := []byte(`{"metadata": {"labels": {"key": "42"}}}`)
+			err = tests.DryRunPatch(restClient, resource, vmim.Name, vmim.Namespace, types.MergePatchType, patch, nil)
+			Expect(err).To(BeNil())
+
+			By("Check that no update actually took place")
+			vmim, err = virtClient.VirtualMachineInstanceMigration(vmim.Namespace).Get(vmim.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vmim.Labels["key"]).ToNot(Equal("42"))
+		})
+	})
+
+	Context("VMI Presets", func() {
+		var preset *v1.VirtualMachineInstancePreset
+		resource := "virtualmachineinstancepresets"
+		presetLabelKey := "kubevirt.io/vmi-preset-test"
+		presetLabelVal := "test"
+
+		BeforeEach(func() {
+			preset = newVMIPreset("test-vmi-preset", presetLabelKey, presetLabelVal)
+		})
+
+		It("create a VMI preset", func() {
+			By("Make a Dry-Run request to create a VMI preset")
+			err = tests.DryRunCreate(restClient, resource, preset.Namespace, preset, nil)
+			Expect(err).To(BeNil())
+
+			By("Check that no VMI preset was actually created")
+			_, err = virtClient.VirtualMachineInstancePreset(preset.Namespace).Get(preset.Name, metav1.GetOptions{})
+			Expect(errors.IsNotFound(err)).To(BeTrue())
+		})
+
+		It("delete a VMI preset", func() {
+			By("Create a VMI preset")
+			_, err := virtClient.VirtualMachineInstancePreset(preset.Namespace).Create(preset)
+			Expect(err).To(BeNil())
+
+			By("Make a Dry-Run request to delete a VMI preset")
+			deletePolicy := metav1.DeletePropagationForeground
+			opts := metav1.DeleteOptions{
+				DryRun:            []string{metav1.DryRunAll},
+				PropagationPolicy: &deletePolicy,
+			}
+			err = virtClient.VirtualMachineInstancePreset(preset.Namespace).Delete(preset.Name, &opts)
+			Expect(err).To(BeNil())
+
+			By("Check that no VMI preset was actually deleted")
+			_, err = virtClient.VirtualMachineInstancePreset(preset.Namespace).Get(preset.Name, metav1.GetOptions{})
```

Comment:
I'm not completely familiar with the api here, but shouldn't we check that the return value is not `nil`? Or is that sure if no error occurs?

Reply:
Yes if `err != nil` it means that the preset exists otherwise something is wrong, so in this case no need to check it because we're not interested in its content but rather in whether it exists or not.

Reply:
OK, thanks!